### PR TITLE
refactor: upgrade lumos to 0.21

### PR DESCRIPTION
**Pull request by homura**
This PR upgraded lumos to 0.21 to work with the `fetch` API

### CODE REVIEW

File: packages/neuron-wallet/src/utils/ckb-rpc.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import type { ParamsFormatter } from '@ckb-lumos/rpc'
 import type { Block } from '@ckb-lumos/base'
 import { HexString } from '@ckb-lumos/base'
 import { CKBRPC } from '@ckb-lumos/rpc'
-import { Method } from '@ckb-lumos/rpc/lib/method'
+import { Method as Method_ } from '@ckb-lumos/rpc/lib/method'
```
Comment (Keith-CY) [inline packages/neuron-wallet/src/utils/ckb-rpc.ts:5]: Does `Method_` conform to a specific naming convention?

File: packages/neuron-wallet/src/services/transaction-sender.ts [hunk 2]
```diff
@@ -855,7 +855,9 @@ export default class TransactionSender {
     const rpc = generateRPC(currentNetwork.remote, currentNetwork.type)
 
     let tx = await rpc.getTransaction(depositOutPoint.txHash)
-    if (tx.txStatus.status !== 'committed') throw new Error('Transaction is not committed yet')
+    if (tx.txStatus.status !== 'committed' || !tx.txStatus.blockHash) {
```
[the remark this answers is not in the document]
Comment (homura) [reply packages/neuron-wallet/src/services/transaction-sender.ts:858]: I am not sure. But I think we'd better stay the same type description with the [rpc-types](https://github.com/nervosnetwork/ckb/blob/860d56c94c386a5d82be6d7f9f50e38420892f88/util/jsonrpc-types/src/blockchain.rs#L584-L591)

**Review by yanguoyu — APPROVED** [review]

File: packages/neuron-wallet/src/services/transaction-sender.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Keith-CY) [reply packages/neuron-wallet/src/services/transaction-sender.ts:858]: Reasonable

File: packages/neuron-wallet/src/utils/ckb-rpc.ts (hunk 1 above)
Comment (homura) [reply packages/neuron-wallet/src/utils/ckb-rpc.ts]: i've renamed to `SdkRpcMethod` for more understandable

**Review by Keith-CY — APPROVED** [review]

File: packages/neuron-wallet/src/utils/ckb-rpc.ts [hunk 3]
```diff
@@ -11,11 +11,11 @@ import {
   PayloadInBatchException,
   IdNotMatchedInBatchException,
 } from '@ckb-lumos/rpc/lib/exceptions'
-import https from 'https'
-import http from 'http'
 import { request } from 'undici'
 import CommonUtils from './common'
 import { NetworkType } from '../models/network'
+import { RPCConfig } from '@ckb-lumos/rpc/lib/types/common'
+import { CKBComponents } from '@ckb-lumos/rpc/lib/types/api'
```
Comment (Keith-CY) [inline packages/neuron-wallet/src/utils/ckb-rpc.ts:18]: These two imports should be marked as `import type` and moved to the top

File: packages/neuron-wallet/src/utils/ckb-rpc.ts [hunk 4]
```diff
@@ -131,6 +131,12 @@ export class FullCKBRPC extends CKBRPC {
   }
 }
 
+class Method extends SdkRpcMethod {
```
Comment (Keith-CY) [inline packages/neuron-wallet/src/utils/ckb-rpc.ts:134]: Maybe `SdkRpcMethod` can be used directly

**Comment by Keith-CY** [comment]
type `Node`(https://github.com/nervosnetwork/neuron/blob/develop/packages/neuron-wallet/src/types/ckbComponents.d.ts#L47-L48) could be updated because `httpAgent` and `httpsAgent` are removed

File: packages/neuron-wallet/src/utils/ckb-rpc.ts (hunk 4 above)
Comment (homura) [reply packages/neuron-wallet/src/utils/ckb-rpc.ts:135]: the super bond a `rpcConfig` as `super(node, options, rpcConfig)`, so we use an extended class here

File: packages/neuron-wallet/src/utils/ckb-rpc.ts (hunk 3 above)
Comment (homura) [reply packages/neuron-wallet/src/utils/ckb-rpc.ts]: switched to `import type`

**Review by Keith-CY — APPROVED** [review]

**Pull request merged**